src/middlewared/middlewared/alert/source/ups.py

```diff
@@ -0,0 +1,43 @@
+from middlewared.alert.base import SimpleOneShotAlertClass, AlertClass, AlertCategory, AlertLevel
+
+
+class UPSBatteryLowAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.UPS
+    level = AlertLevel.ALERT
+    title = 'UPS Battery LOW'
+    text = 'UPS %(ups)s Battery level low.'
+
+
+class UPSOnlineAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.UPS
+    level = AlertLevel.INFO
+    title = 'UPS ONLINE'
+    text = 'UPS %(ups)s is on line power.'
+
+
+class UPSOnBatteryAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.UPS
+    level = AlertLevel.CRITICAL
+    title = 'UPS On Battery'
+    text = 'UPS %(ups)s is on battery power.'
+
+
+class UPSCommbadAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.UPS
+    level = AlertLevel.CRITICAL
+    title = 'UPS Communication lost'
+    text = 'Communications with UPS %(ups)s lost.'
+
+
+class UPSCommokAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.UPS
+    level = AlertLevel.INFO
+    title = 'UPS Communications established'
+    text = 'Communications with UPS %(ups)s established.'
+
+
+class UPSReplbattAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.UPS
+    level = AlertLevel.CRITICAL
+    title = 'UPS Battery needs to be replaced'
```

Comment:
We should be using title case for every word
USB Battery Needs To Be Replace

Thats the unspoken rule we have been using, iirc

Reply:
Yes, these should be title case.  But title case is not every word, there are odd rules.
There are online things that will do it correctly, mostly.